mbed_build/_internal/find_files.py

```diff
@@ -51,3 +89,10 @@ def _matches_mbedignore_patterns(path: Path, patterns: Iterable[str]) -> bool:
     """Check if given path matches one of the .mbedignore patterns."""
     stringified = str(path)
     return any(fnmatch(stringified, pattern) for pattern in patterns)
```

Comment:
Unless I am mistaken fnmatch only works for files such as `*.py` or `test*`
However, wouldn't mbedignore include patterns on folders such as `test/**` I would suggest the use of https://facelessuser.github.io/wcmatch/wcmatch/ which I use in the CI scripts

Reply:
Old tools use `fnmatch`. I wasn't able to find any docs about the spec of `.mbedignore` files, only lots of issues on GitHub. 😞 

I'd rather not change how it works, as it might have unexpected side effects.

Fwiw, list of all `.mbedignore` rules from `mbed-os-example-blinky` tree:
```
*
*lwip/doc/*
lwip/test/*
lwip/src/apps/*
lwip/src/netif/lwip_slipif.c
lwip/src/include/lwip/apps/*
lwip/src/include/compat/stdc/*
lwip-eth/*
*
cs_nvm/test/*
test/*
unittest/*
release/*
source/Service_Libs/CCM_lib/mbedOS/aes_mbedtls.c
output/*
test/*
test/*
lfs_emubd.*
littlefs/emubd/
littlefs/tests/
TESTS/util
tests/*build/*
yotta_modules/*
yotta_targets/*
test/*
example/*
test/*
unittest/*
test/*
linux/*
test/*
*test/*
tests/*
```

Reply:
I was more reasoning as a linux person writing the .mbedignore

Reply:
`fnmatch` works with unix shell `glob` patterns which allow for recursive globbing.

```
>>> fnmatch.fnmatch("blah/hah/moo/boo", "**/boo")
True
```